yt_dlp/extractor/cspan.py

```diff
@@ -242,3 +246,52 @@ def get_text_attr(d, attr):
                 'title': title,
                 'id': 'c' + video_id if video_type == 'clip' else video_id,
             }
+
+
+class CSpanCongressIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?c-span\.org/congress/'
+    _TESTS = [{
+        'url': 'https://www.c-span.org/congress/?chamber=house&date=2017-12-13&t=1513208380',
+        'info_dict': {
+            'id': 'house_2017-12-13',
+            'title': 'Congressional Chronicle - Members of Congress, Hearings and More',
+            'description': 'md5:54c264b7a8f219937987610243305a84',
+            'thumbnail': r're:https://ximage.c-spanvideo.org/.+',
+            'ext': 'mp4'
+        }
+    }, {
+        'url': 'https://www.c-span.org/congress/',
+        'info_dict': {
+            'id': 'congress',
+            'title': 'Congressional Chronicle - Members of Congress, Hearings and More',
+            'description': 'md5:54c264b7a8f219937987610243305a84',
+            'thumbnail': r're:https://ximage.c-spanvideo.org/.+',
+            'ext': 'mp4'
+        }
+    }]
```

Comment:
This is a problem. The video id MUST be both unique and consistent (per extractor). ie, (1) Diff videos must never have the same id and, (2) A video must always have the same id even if download using different URLs. This does not seem to satisfy either case

Reply:
Regarding the first question:

```
yt-dlp "https://www.c-span.org/video/?517228-1/senate-debates-russian-sanctions-voting-rights-legislation" -v

[debug] Command-line config: ['https://www.c-span.org/video/?517228-1/senate-debates-russian-sanctions-voting-rights-legislation', '-v']
[debug] Encodings: locale cp1251, fs utf-8, out utf-8, err utf-8, pref cp1251
[debug] yt-dlp version 2021.12.27 [6223f67a8]
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.8.7 (CPython 64bit) - Windows-10-10.0.19041-SP0
[debug] exe versions: ffmpeg 4.4-full_build-www.gyan.dev (setts), ffprobe 4.4-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome, mutagen, sqlite, websockets
[debug] Proxy map: {}
[debug] [CSpan] Extracting URL: https://www.c-span.org/video/?517228-1/senate-debates-russian-sanctions-voting-rights-legislation
[CSpan] 517228: Downloading webpage
[CSpan] 517228: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, vbr, abr, asr, proto, vext, aext, hasaud, source, id
ERROR: matching opening tag for closing h2 tag not found
Traceback (most recent call last):
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\YoutubeDL.py", line 1365, in wrapper
    return func(self, *args, **kwargs)
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\YoutubeDL.py", line 1435, in __extract_info
    ie_result = ie.extract(url)
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\extractor\common.py", line 609, in extract
    ie_result = self._real_extract(url)
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\extractor\cspan.py", line 129, in _real_extract
    title = get_element_by_class('video-page-title', webpage) or \
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\utils.py", line 1964, in get_element_by_class
    retval = get_elements_by_class(class_name, html)
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\utils.py", line 1986, in get_elements_by_class
    return get_elements_by_attribute(
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\utils.py", line 2000, in get_elements_by_attribute
    return [content for content, _ in get_elements_text_and_html_by_attribute(*args, **kwargs)]
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\utils.py", line 2000, in <listcomp>
    return [content for content, _ in get_elements_text_and_html_by_attribute(*args, **kwargs)]
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\utils.py", line 2025, in get_elements_text_and_html_by_attribute
    content, whole = get_element_text_and_html_by_tag(m.group('tag'), html[m.start():])
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\utils.py", line 2102, in get_element_text_and_html_by_tag
    parser.feed(html[offset:offset + next_closing_tag_end])
  File "c:\program files\python38\lib\html\parser.py", line 111, in feed
    self.goahead(0)
  File "c:\program files\python38\lib\html\parser.py", line 173, in goahead
    k = self.parse_endtag(i)
  File "c:\program files\python38\lib\html\parser.py", line 421, in parse_endtag
    self.handle_endtag(elem)
  File "c:\work\projects\grabien\yt-dlp\yt_dlp\utils.py", line 2070, in handle_endtag
    raise compat_HTMLParseError(f'matching opening tag for closing {tag} tag not found')
yt_dlp.compat.compat_HTMLParseError: matching opening tag for closing h2 tag not found
```